### ISSUE TITLE
Support Ruby's handling of single quotes and double quote strings

### ISSUE THREAD

**Issue by AlanFoster** [iamsolankiamit/prettier-ruby#20]
The semantics of single and double quotes are different in Ruby to Python/JavaScript, let's ensure we align with the semantics that would be expected with Ruby.

## Input

```ruby
'hello world'
"hello world"

'#{not interpolated}'

'hello\nworld1'
"hello\nworld2"

'\n'
"\\n"
"\n"

'"'
"'"

'\\'
"\\"
```

## Output

I believe this is the output we would want for the above input:

```ruby
'hello world'
'hello world'

'#{not interpolated}'

'hello\nworld1'
"hello\nworld2"

'\n'
'\n'
"\n"

'"'
"'"

'\\'
'\'
```

---

Note this work is blocked until tokens can be correlated with sexpressions 🤞

**Comment by AlanFoster** [comment]
This should handle empty strings too, currently this crashes prettier ruby:

```ruby
''
""
```

**Comment by iamsolankiamit** [comment]
Fixed in (d0b9c50) and (0dc7555) .
